### ISSUE TITLE
Allow splitting the integrated terminal in both directions

### ISSUE THREAD

**Issue by mneil** [microsoft/vscode#56112]
I was unable to find another request like this. Most search results talk about wanting to dock the terminal/panel to the right or the bottom. 

It would be nice if, when using the integrated terminal and splitting the terminal, I could choose to stack the terminals (one on top of the other) or put them side by side. Right now it seems like every time I split the terminal is puts the newest one to the right. 

The option to have new terminals go underneath the last one would be great. I'm using a monitor rotated 90 degrees and I have a lot of real-estate at the bottom of the monitor where I can stack multiple terminals. I know it's not uncommon for developers to have 1 monitor rotated for development and I hope this would be helpful for other users as well.

**Comment by Tyriar** [comment]
This was a decision made when adding splitting to simplify the solution as well as the UX (by only allowing splitting horizontally when at bottom or vertically when at the right). I consider any additional movement on this blocked on https://github.com/Microsoft/vscode/issues/10121 (where we could potentially pull the panel out of the window).

**Comment by ndbeals** [comment]
@Tyriar do you guys even have any plans for the Floating Window Request? It's sat on hold for over 2 years.

If nothing is going to happen there, then nothing is going to happen for this request either.

**Comment by mneil** [comment]
I'm not even sure floating windows solve the problem. If I float the console and split it then I still have the same problem unless you add this functionality.

Or, if you expect me to open multiple windows each with a console I would be responsible for their layout. The ux would not be the same if I were left to resize and move multiple windows independently.

While I think it's related to pop out windows I don't know that it's dependent on that issue. For me, personally, I wouldn't even need more ui. A simple editor config to set whether the windows stack or or go side by side would be enough.

**Comment by Tyriar** [comment]
@ndbeals the issue mentions there's a technical limitation. I certainly hope we do soon as it will enable a lot of cool stuff.

@mneil this isn't _technically_ blocked on it, but only when floating windows are supported, and you can pull the panel out, would we be able to justify spending time on this issue.

**Comment by FalseProtagonist** [comment]
I am trying to migrate from emacs and this is the second thing I tried to work out how to do. 

How does vscode "think" about windows/frames/buffers? Is there a deep distinction with editors vs panels? Are windows/frames/buffers exposed such that I could write an extension to implement this?

**Comment by Tyriar** [comment]
@FalseProtagonist currently editors and panels and viewlets are all very different things, you wouldn't be able to write an extension to do this.

**Comment by aneilbaboo** [comment]
I'd really like to see this too. I usually have about 3 terminals open. One for my backend, one for my front end, and another to interact with. I'd like to be able to do this:

<table>
    <tbody>
        <tr>
            <td rowspan="2">Interactive Terminal</td>
            <td colspan="1">Frontend terminal</td>
        </tr>
        <tr>
            <td>Backend Terminal</td>
        </tr>
    </tbody>
</table>

Having to put all three into a horizontal row doesn't provide enough width on a typical 15" laptop.

**Comment by Jay-Schneider** [comment]
I found this issue on my search for how to split the integrated terminal vertically when it is at the bottom. I was surprised to see that this still seems to be impossible and also would be interested in having this feature.

Just for all others finding this while looking for a solution: I now workaround it by starting tmux in my rightmost terminal to achieve a setup as @aneilbaboo describes it.

**Comment by heathsnow** [comment]
One workaround is to install `screen` (On Mac: `brew "screen"`).  Then type `screen` + Enter, ctl + a, shift + s, ctl + a, tab, ctl +a, c.  Then use ctl + tab to shift back and forth.

**Comment by Tyriar** [comment]
No plans to implement this for the panel specifically, however we do want to bring terminals to the editor area which would enable this use case. Terminal tab issue here: https://github.com/microsoft/vscode/issues/10546
